ais_service_discovery/services/cloudmap.py

```diff
@@ -4,14 +4,15 @@ def __init__(self, client):
 
     def filter_instances(self, response, id):
         filter_function = lambda instance : instance['InstanceId'] == id
-
+        
         return {
             **response,
-            'Instances': filter(filter_function, response['Instances'])
+            'Instances': list(filter(filter_function, response['Instances']))
```

Comment:
`filter` function by default returns an iterable list here, anything specific explicit type casting solves here?

Reply:
It was throwing an error for me when no instances were returned. It was returning an empty <filter> object, but we were then trying to pop a value out of it, which was throwing a 'filter not subscribable' error